packages/ui-extensions/src/surfaces/checkout/api/shared.ts

```diff
@@ -148,3 +148,34 @@ export interface ShippingAddress extends MailingAddress {
    */
   oneTimeUse?: boolean;
 }
+
+type MaybePromise<T> = T extends Promise<any> ? T : T | Promise<T>;
+type Subscriber<T> = (value: T) => unknown;
+type RemoteSubscribeResult<T> = [() => void, T];
+
+/**
+ * A remote-subscribable object managed on the main thread.
+ *
+ * Example: Checkout uses this to manage the state of an object and
+ * communicate state changes to extensions running in a sandboxed web worker.
+ */
+export interface RemoteSubscribable<T> {
+  readonly initial: T;
+  subscribe(subscriber: Subscriber<T>): MaybePromise<RemoteSubscribeResult<T>>;
+}
```

Comment:
This expresses a subscribable on the main thread. It's irrelevant to the public `ui-extensions` library and I think it was a mistake to move it into the library in https://github.com/shop/world/pull/127597 (private link).  We should delete it and move it back to the checkout codebase which is the only place it's needed.